Modules/Sources/WordPressReader/Comments/Views/WebCommentContentRenderer.swift

```diff
@@ -0,0 +1,158 @@
+import WebKit
+import WordPressShared
+import WordPressUI
+
+/// Renders the comment body with a web view. Provides the best visual experience but has the highest performance cost.
+@MainActor
+public final class WebCommentContentRenderer: NSObject, CommentContentRenderer {
+    // MARK: Properties
+
+    public weak var delegate: CommentContentRendererDelegate?
+
+    private let webView = WKWebView(frame: .zero, configuration: {
+        let configuration = WKWebViewConfiguration()
+        configuration.allowsInlineMediaPlayback = true
+        configuration.defaultWebpagePreferences.allowsContentJavaScript = false
+        return configuration
+    }())
+
+    private var comment: String?
+
+    /// It can't be changed at the moment, but this capability was included from the
+    /// start, and this implementation continues supporting it.
+    private var displaySetting = ReaderDisplaySettings.standard
+
+    /// - warning: This has to be configured _before_ you render.
+    public var tintColor: UIColor {
+        get { webView.tintColor }
+        set {
+            webView.tintColor = newValue
+            cachedHead = nil
+        }
+    }
+
+    private var cachedHead: String?
+
+    // MARK: Methods
+
+    public required override init() {
+        super.init()
+
+        if #available(iOS 16.4, *) {
+            webView.isInspectable = true
+        }
+        webView.backgroundColor = .clear
+        webView.isOpaque = false // gets rid of the white flash upon content load in dark mode.
+        webView.translatesAutoresizingMaskIntoConstraints = false
+        webView.navigationDelegate = self
+        webView.scrollView.bounces = false
+        webView.scrollView.showsVerticalScrollIndicator = false
+        webView.scrollView.backgroundColor = .clear
+    }
+
+    public func render(comment: String) -> UIView {
+        guard self.comment != comment else {
+            return webView // Already rendering this comment
+        }
+        self.comment = comment
+
+        // - important: `wordPressSharedBundle` contains custom fonts
+        webView.loadHTMLString(formattedHTMLString(for: comment), baseURL: Bundle.wordPressSharedBundle.bundleURL)
+        return webView
+    }
+}
+
+// MARK: - WKNavigationDelegate
+
+extension WebCommentContentRenderer: WKNavigationDelegate {
+    public func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        // Wait until the HTML document finished loading.
+        // This also waits for all of resources within the HTML (images, video thumbnail images) to be fully loaded.
+        webView.evaluateJavaScript("document.readyState") { complete, _ in
```

Comment:
This was the existing code. I just moved it. Not sure if it works the way it says it does.